src/lsp/cobol_parser/grammar.mly

```diff
@@ -2894,15 +2913,33 @@ let accept_misc :=
  | EXCEPTION; STATUS;      {AcceptExceptionStatus} (* MF *)
 
 let accept_clause :=
- | ~ = at_position;                    <AcceptAt>
- | FROM; CRT;                          {AcceptFromCRT}   (* MF *)
- | MODE; IS?; BLOCK;                   {AcceptModeBlock} (* MF *)
- | WITH; ~ = loc(accept_with_clause)+; <AcceptWith>      (* MF *)
-
-let accept_with_clause :=
- | UPDATE;                      {AcceptUpdate}
- | ~ = screen_attribute_clause; <AcceptAttribute>
-(* TODO: MF allows many clauses *)
+  | ~ = at_position;                                  <AcceptAt>
+  | FROM; CRT;                                        {AcceptFromCRT}   (* MF *)
+  | MODE; IS?; BLOCK;                                 {AcceptModeBlock} (* MF *)
+  | WITH?; ~ = nell(loc(accept_with_clause)); %prec lowest <AcceptWith> (* MF *)
+    (* Note: GnuCOBOL makes "WITH" optional *)
+
+let accept_with_clause := (* MF *)
```

Comment:
Here you could do:
```ocaml
let accept_with_clause [@symbol "<accept-with-clause>"] :=
```
to get better syntax hints.

Same for `display_with_clause`.

Reply:
How does that work ?

Reply:
When an `accept_with_clause` is actually required as per the grammar, the recovery mechanism uses that `<accept-with-clause>` string instead of a possibly arbitrary sequence of tokens that forms a valid `accept_with_clause`. The auto-completed/missing sequence of tokens may be selected less arbitrarily by using `[@cost]` annotations, but adding the suggested string makes the sequence more readable.

Reply:
@engboris I think that aspect is interesting for you as well (I planned to explained that last week, but somehow I lacked the opportunity to do so).